lib/widgets/content.dart

```diff
@@ -432,20 +432,34 @@ class MessageUnicodeEmoji extends StatelessWidget {
   }
 }
 
-class MessageRealmEmoji extends StatelessWidget {
-  const MessageRealmEmoji({super.key, required this.node});
+class MessageImageEmoji extends StatelessWidget {
+  const MessageImageEmoji({super.key, required this.node});
 
-  final RealmEmojiNode node;
+  final ImageEmojiNode node;
 
   @override
   Widget build(BuildContext context) {
-    // TODO show actual emoji image
-    final alt = node.alt;
-    return Container(
-        padding: const EdgeInsets.all(2),
-        decoration: BoxDecoration(
-            color: Colors.white, border: Border.all(color: Colors.purple)),
-        child: Text(alt));
+    final store = PerAccountStoreWidget.of(context);
+    final adjustedSrc = rewriteImageUrl(node.src, store.account);
+
+    const size = 20.0;
+
+    return Stack(
+        alignment: Alignment.center,
+        clipBehavior: Clip.none,
+        children: [
+          const SizedBox(width: size, height: kBaseFontSize),
+          Positioned(
+              // Web's css makes this seem like it should be -0.5, but that looks
+              // too low.
+              top: -1.5,
+              child: Image.network(
+                adjustedSrc.toString(),
+                filterQuality: FilterQuality.low,
```

Comment:
This should be `FilterQuality.medium`, same as for avatars and attached images.

At `low`, it… actually it looks pretty much fine in your screenshot from iOS, but here's what it is for me on Linux desktop:
![image](https://user-images.githubusercontent.com/28173/210032154-e8586042-cd77-4a67-b92d-a29d7dcb463c.png)
Pretty pixelated and jagged.

At `medium`, it looks much better, and pretty similar to what web gets in Chrome on Linux:
![image](https://user-images.githubusercontent.com/28173/210032217-bb90b085-c2c3-4c19-8e37-1a142e43d2de.png)
(left is Flutter desktop, right is Chrome)

… OK, and trying it on Android, on a physical Pixel 5, both `low` and `medium` look good; in fact I can't tell them apart, at least not on this emoji at this size. It seems possible that on that device they actually end up selecting the exact same algorithm.

In your testing, do you find `low` is actually better than `medium`? If so, perhaps we should choose different algorithms on different platforms.

In any case we almost surely want the same `filterQuality` value here as in sender avatars, and likely also in attached images. So let's keep it in sync with those.

Reply:
> In your testing, do you find `low` is actually better than `medium`? If so, perhaps we should choose different algorithms on different platforms.

No, I just figured that emojis are probably close to the smallest images one would want to render, and so `low` would be appropriate. Seeing that it looks worse than `medium` on some platforms, and that we don't have anything close to a performance problem yet, I'm happy to use `medium`. 🙂